.github/workflows/update-flake-dependencies.yml

```diff
@@ -0,0 +1,37 @@
+# CI job to periodically (once a week) update flake.lock
+name: Update flake dependencies
```

Comment:
I mostly copied it out of my own dotfiles repo 😛 

Just the bit to create the PR is a little different because we require signed commits in this repo.